security_token_approval.teal

```diff
@@ -493,4 +493,84 @@ btoi
 app_local_put
 int 1
 return
-l12:
\ No newline at end of file
+b l13
```

Comment:
@aquabu looks like we are exceeding the 1000 bytes limit after adding the function, since under the hood TEAL still  duplicates the restrictions checking logic twice.

Current size is 1098 bytes.

Reply:
Maybe we can try to solve this by unifying the indices, but then `detectTransferRestriction` could only be called from the same account that would be performing the transfer.

Reply:
In good news: 
Extracting the transfer restrictions logic into a helper method seems to be working just fine (:

So it's only app size issue.

Reply:
Cool. I extracted some other logic for an updated implementation of totalSupply that functions more like the ERC20 token (doesn't include tokens that haven't been minted from the reserve).